vmware-event-router/internal/processor/processor.go

```diff
@@ -11,3 +13,21 @@ import (
 type Processor interface {
 	Process(types.ManagedObjectReference, []types.BaseEvent) error
 }
+
+// Error struct contains the generic error content used by the processors
+// it extends the simple error by providing context which processor gave
+// the error
+type Error struct {
+	processor string
+	err       error
+}
+
+// NewError constructs Error structure
+func NewError(processor string, err error) *Error {
+	return &Error{
+		processor: processor,
+		err:       err,
+	}
+}
+
+func (e *Error) Error() error { return fmt.Errorf("%s: %s", e.processor, e.err.Error()) }
```

Comment:
We can simply satisfy the error interface:

```go
type error interface {
    Error() string
}
```

Reply:
You are right, will replace 👍

Reply:
Returns string (error interface) ✔️